### PR TITLE
Documentation and examples for autostarting with Upstart

### PR DESCRIPTION
**Pull request by acogdev**
Added documentation and examples for Ubuntu-like linux systems using Upstart as the init system in response to syncthing/syncthing#2315.

### CODE REVIEW

File: etc/linux-upstart/system/syncthing.conf [hunk 1]
```diff
@@ -0,0 +1,13 @@
+description "Syncthing"
+
+start on (local-filesystems and net-device-up IFACE!=lo)
+stop on runlevel [!2345]
+
+env STNORESTART=yes
+env HOME=/home/jake
```
Comment (AudriusButkevicius) [inline etc/linux-upstart/system/syncthing.conf]: Can this be $USER?

Comment (acogdev) [inline etc/linux-upstart/system/syncthing.conf]: Yes, it should be $USER.

**Comment by acogdev** [comment]
I fixed etc/linux-upstart/system/syncthing.conf  to use $USER

**Comment by AudriusButkevicius** [comment]
I'll leave for @calmh to test as I don't use upstart.

**Comment by calmh** [comment]
@st-jenkins add to whitelist

**Comment by calmh** [comment]
LGTM. Next time, add a `(fixes 2315)` to the commit subject as well to get the issue closed automatically on merge.

**Comment by calmh** [comment]
Thanks!

**Pull request merged**